system/repositories.rs

```diff
@@ -0,0 +1,112 @@
+pub mod user_repository {
+    use crate::concepts::{Credential, PasswordHash, User};
+    use crate::util::query_util::{execute_read_pipeline, execute_write_pipeline};
+    use answer::variable_value::VariableValue;
+    use concept::thing::thing_manager;
+    use concept::type_::type_manager::TypeManager;
+    use database::transaction::{TransactionRead, TransactionWrite};
+    use function::function_manager::FunctionManager;
+    use std::collections::HashMap;
+    use std::sync::Arc;
+    use storage::durability_client::WALClient;
+    use storage::snapshot::WriteSnapshot;
+    use thing_manager::ThingManager;
+    use typeql::parse_query;
+    use uuid::Uuid;
+    use crate::util::answer_util::get_string;
+
+    pub fn list(tx: TransactionRead<WALClient>) -> Vec<User> {
+        let query = parse_query(
+            "match (user: $u, password: $p) isa user-password; $u has name $n;"
+        ).unwrap();
+        let (tx, result) =
+            execute_read_pipeline(tx, &query.into_pipeline());
+        let rows = result.unwrap();
+        let mut users: Vec<User> = vec![];
```

Comment:
These 5 lines 25 - 30 can be rewritten as a single `rows.map(|row| ...).collect()`

Reply:
Fixed in 8910c738519f0fbc7618e98c9dea3b9440935462.